### PR TITLE
Add more commit-attributes to `Merge`

### LINKED ISSUE #5995: Support setting merge author in API v2
Nessie allows having separate author and committer names for normal commits.

However, the merge API does not currently allow setting the author and the committer is reused as the author on the server side.

In API v2 all merges result is one commit being made on the target branch (squashed). So, it makes sense to allow the client to provide a separate (optional) author name for that merge commit.

### PR DESCRIPTION
**Pull request by snazy**
The new property `Merge.getCommitMeta()` allows specifying more commit attributes than just the message:
* message
* authors
* author-time
* signed-off-by
* properties

Fixes #5995

### CODE REVIEW

**Review by dimas-b — COMMENTED** [review]
LGTM overall :+1:

File: servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieApi.java [hunk 1]
```diff
@@ -475,9 +476,48 @@ public void commitMergeTransplant() throws Exception {
...
+              CommitMeta::getAllAuthors,
+              CommitMeta::getAllSignedOffBy,
+              CommitMeta::getAuthorTime,
+              CommitMeta::getProperties)
+          .containsExactly(
+              "My custom merge message",
+              singletonList("NessieHerself"),
+              singletonList("Arctic"),
+              Instant.EPOCH,
+              ImmutableMap.of("property", "value", "_merge_parent", branch.getHash()));
```
[the remark this answers is not in the document]
Comment (snazy) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieApi.java:514]: Becuase it's returned :(

File: servers/rest-services/src/main/java/org/projectnessie/services/rest/RestV2TreeResource.java [hunk 2]
```diff
@@ -340,14 +347,28 @@ public MergeResponse transplantCommitsIntoBranch(String branch, Transplant trans
   public MergeResponse mergeRefIntoBranch(String branch, Merge merge)
       throws NessieNotFoundException, NessieConflictException {
     ParsedReference ref = resolveRef(branch);
+
+    String msg = merge.getMessage();
+
+    ImmutableCommitMeta.Builder meta = CommitMeta.builder();
+    CommitMeta commitMeta = merge.getCommitMeta();
+    if (commitMeta != null) {
+      meta.from(commitMeta).committer(null).commitTime(null).parentCommitHashes(emptyList());
```
[the remark this answers is not in the document]
Comment (dimas-b) [reply servers/rest-services/src/main/java/org/projectnessie/services/rest/RestV2TreeResource.java:357]: I see that we have `validateCommitMeta()` in `TreeApiImpl` now, but if we null-out `committer` and `commitTime` here, aren't we losing that validation on this call path?

[the remark this answers is not in the document]
Comment (snazy) [reply servers/rest-services/src/main/java/org/projectnessie/services/rest/RestV2TreeResource.java:357]: good point (not a serious thing - just missing the validation)

updated

**Review by dimas-b — APPROVED** [review]

**Pull request merged**